framework/lib/components/createable-select-dropdown/createable-select-dropdown.tsx

```diff
@@ -78,95 +79,98 @@ import { customComponents } from './custom-components'
+    return targetValue
+      ? standardOptions.find((option) => option.value === targetValue) ?? null
+      : null
+  })
 
-  const [ selectedOption, setSelectedOption ] =
-    useState<Option | null>(initialSelectedOption)
-  const [ newOptionText, setNewOptionText ] = useState('')
-  const [ newOptionPlaceholder, setNewOptionPlaceholder ] = useState(initialPlaceholder)
+  const [ newOptionPlaceholder, setNewOptionPlaceholder ] =
+    useState(initialPlaceholder)
   const [ createdOptions, setCreatedOptions ] = useState<Option[]>([])
+  const [ createNewOption, setCreateNewOption ] = useState<boolean>(false)
+  const [ addFieldInputValue, setAddFieldInputValue ] = useState<string>('')
+
+  const toggleCreateNewOption = () => {
+    setCreateNewOption(not)
+  }
 
   function isCreationOption (option: any): option is CreationOption {
     return option && typeof option.isCreation === 'boolean'
   }
 
   const ref = useRef<HTMLDivElement>(null)
-  useOutsideClick({
-    ref,
-    handler: () => {
-      if (isCreationOption(selectedOption)) {
-        setSelectedOption(initialSelectedOption)
-        setNewOptionPlaceholder(initialPlaceholder)
-      }
-    },
-  })
 
-  const handleInputChange = (newValue: string) => {
-    setNewOptionText(newValue)
-  }
+  const handleCreateOption = (newOptionTextString: string) => {
+    toggleCreateNewOption()
+
+    if (isEmpty(trim(newOptionTextString))) {
+      return
+    }
 
-  const handleCreateOption = () => {
-    const newOptionTextLower = newOptionText.toLowerCase()
-    const newOptionExists = standardOptions.some((option) =>
-      option.value.toLowerCase() === newOptionTextLower)
+    const newOptionTextLower = R.toLower(newOptionTextString)
 
-    if (!newOptionExists) {
-      const newOption = { label: newOptionText, value: newOptionText }
-      setCreatedOptions((currentCreatedOptions) => [ ...currentCreatedOptions, newOption ])
+    const newOptionExistsInStandardOptions = standardOptions.some(
+      (option) => option.value.toLowerCase() === newOptionTextLower
+    )
+    const newOptionExistsInCreatedOptions = createdOptions.some(
+      (option) => option.value.toLowerCase() === newOptionTextLower
+    )
+
+    if (!newOptionExistsInCreatedOptions && !newOptionExistsInStandardOptions) {
+      const newOption = {
+        label: newOptionTextString,
+        value: newOptionTextString,
+      }
+      setCreatedOptions((currentCreatedOptions) => [
+        ...currentCreatedOptions,
+        newOption,
+      ])
       setSelectedOption(newOption)
       onOptionChange(newOption)
     }
-
-    setNewOptionText('')
     setNewOptionPlaceholder(initialPlaceholder)
   }
 
   const handleChange = (
     newValue: SingleValue<Option>,
-    _actionMeta: ActionMeta<Option>) => {
+    _actionMeta: ActionMeta<Option>
+  ) => {
     const option = newValue
 
-    if (option === null) {
+    if (newValue?.isCreation) {
+      toggleCreateNewOption()
+    }
```

Comment:
@GustafGronlund I think these lines of code are the culprit to the problem in my previous comment. When you click on the 'Add field' it triggers toggleCreateNewOption in custom-components.tsx, but since you also call it in handleChange it ends up being called twice.